main.tf

```diff
@@ -117,7 +117,7 @@ data "aws_iam_policy_document" "ecs_service" {
 }
 
 resource "aws_iam_role" "ecs_service" {
-  count                = var.enabled ? 1 : 0
+  count                = var.enabled && var.network_mode != "awsvpc" ? 1 : 0
```

Comment:
thanks @nitrocode 
to make it complete, please add
```
count                = var.enabled && var.network_mode != "awsvpc" ? 1 : 0
```

to 

data "aws_iam_policy_document" "ecs_service_policy"
and
resource "aws_iam_role_policy" "ecs_service"